### PR TITLE
rpcsrv: handle preflight OPTIONS with CORS kludge, fix #2721

### LINKED ISSUE #2721: allow preflight OPTIONS request
Would be good to handle preflight request, currently it's hard to query rpc node with browser app, browser tries to do OPTIONS request before POST (it asks server, which request methods are allowed, allowed origins and etc.), but getting
 `{"id":null,"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid Params","data":"invalid method 'OPTIONS', please retry with 'POST'"}}`
It's this part of code responsible for that in `pkg/services/rpcsrv/server.go`, which retrict any methods that not POST
image

Actually what can be done is - return successful response with allow methods (only POST in this case), like described in mdn docs [click](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS)
Doing that will let browser to successfully continue POST request :thinking:

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

Not broken enough CORS.

### Solution

Break more of it.

### CODE REVIEW

File: pkg/services/rpcsrv/server.go [hunk 1]
```diff
@@ -446,6 +446,13 @@ func (s *Server) handleHTTPRequest(w http.ResponseWriter, httpRequest *http.Requ
 		return
 	}
 
+	if httpRequest.Method == "OPTIONS" && s.config.EnableCORSWorkaround { // Preflight CORS.
+		setCORSOriginHeaders(w.Header())
+		w.Header().Set("Access-Control-Allow-Methods", "GET, POST") // GET for websockets.
+		w.Header().Set("Access-Control-Max-Age", "31536000")
```
Comment (AnnaShaleva) [inline pkg/services/rpcsrv/server.go:452]: A year of caching, we are quite stable :D

**Review by AnnaShaleva — COMMENTED** [review]
`EnableCORSWorkaround` is off on all our public nodes (testnet, mainnet). Maybe it's worth to move preflight OPTIONS setting to a separate config option and enable it on public networks?

File: pkg/services/rpcsrv/server.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/services/rpcsrv/server.go:452]: OK, let's have some hours, someone might want to turn these options on and off.

**Comment by roman-khimov** [comment]
> EnableCORSWorkaround is off on all our public nodes (testnet, mainnet)

It's on.

**Comment by AnnaShaleva** [comment]
> > EnableCORSWorkaround is off on all our public nodes (testnet, mainnet)
> 
> It's on.

Ok, I've just looked at our default mainnet/testnet config, it's set to false there.

**Review by AnnaShaleva — APPROVED** [review]

**Comment by roman-khimov** [comment]
Yes, it's off by default (as it should be, it makes CORS useless), but we have it enabled on public servers (because people want to break CORS).

**Pull request merged**